drivers/tee/tee_shm.c

```diff
@@ -20,25 +20,42 @@
 #include <linux/tee_drv.h>
 #include "tee_private.h"
 
+/* extra references appended to shm object for registered shared memory */
+struct dmabuf_ref {
+	struct dma_buf *dmabuf;
+	struct dma_buf_attachment *attach;
+	struct sg_table *sgt;
+};
```

Comment:
I think it would be better to have a struct like:
```
struct tee_shm_dmabuf_ref {
     struct tee_shm;
     struct dma_buf *dmabuf;
     struct dma_buf_attachment *attach;
     struct sg_table *sgt
};
```

and use container_of to go from a pointer to a struct tee_shm to a pointer to a struct tee_shm_dmabuf_ref.